src/test/documentation_examples.rs

```diff
@@ -0,0 +1,1385 @@
+use futures::TryStreamExt;
+
+use crate::{
+    bson::{doc, Bson},
+    error::Result,
+    options::FindOptions,
+    test::TestClient,
+    Collection,
+};
+
+macro_rules! assert_coll_count {
+    ($coll:expr, $expected:expr) => {
+        assert_eq!($coll.count_documents(None, None).await.unwrap(), $expected);
+    };
+}
+
+macro_rules! assert_cursor_count {
+    ($cursor:expr, $expected:expr) => {{
+        let docs: Vec<_> = $cursor.try_collect().await.unwrap();
```

Comment:
Unfortunately, `StreamExt` doesn't have `count` method like [`Iterator` does](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.count). A long time back I tried to convince Lucio that there should be one, but I was sadly unsuccessful.